xonsh/main.py

```diff
@@ -386,6 +372,20 @@ def func_sig_ttin_ttou(n, f):
             code = sys.stdin.read()
             run_code_with_cache(code, shell.execer, glb=shell.ctx, loc=None,
                                 mode='exec')
+        if args.force_interactive:
```

Comment:
Also, why move away from `args.mode`?  Shouldn't args.force_interactive just set `args.mode` to be interactive in premain?